### PR TITLE
Added AbstractSystemProperty and 4 system properties implementation.

### PR DESCRIPTION
**Pull request by SherifWaly**
PR for #201

### CODE REVIEW

**Comment by SherifWaly** [comment]
@amihaiemil Please check this commit :D

File: src/main/java/com/amihaiemil/charles/aws/AbstractSystemProperty.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    
+    /**
+     * Ctor
+     * @param String name.
+     */
+    public AbstractSystemProperty(String name) {
+        this.name = name;
+    }
+
+	@Override
```
Comment (amihaiemil) [inline src/main/java/com/amihaiemil/charles/aws/AbstractSystemProperty.java]: @SherifWaly Indentation to the left here

File: src/main/java/com/amihaiemil/charles/aws/StEsEndPoint.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+public final class StEsEndPoint extends AbstractSystemProperty implements EsEndPoint {
+    
+    /**
+     * Ctor. 
+     * @param String name
+     */
+    public StEsEndPoint(final String name) {
+        super(name);
+    }
+
```
Comment (amihaiemil) [inline src/main/java/com/amihaiemil/charles/aws/StEsEndPoint.java]: @SherifWaly For this, can you override the ``read()`` method, to check if the endpoint ends with "/" or not? If not, then add it. So: 

```
public String read() {
    String endpoint = super.read()
    //... check for '/' here
}

**Review by amihaiemil — COMMENTED** [review]
@SherifWaly Just 2 comments :D

File: src/main/java/com/amihaiemil/charles/aws/StAccessKeyId.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ */
+package com.amihaiemil.charles.aws;
+
+/**
+ * Implementation for {@link AccessKeyId} interface.
+ * @author Sherif Waly (sherifwaly95@gmail.com)
+ * @version $Id$
+ * @since 1.0.0
+ */
+public final class StAccessKeyId extends AbstractSystemProperty implements SecretKey {
```
Comment (amihaiemil) [inline src/main/java/com/amihaiemil/charles/aws/StAccessKeyId.java]: @SherifWaly Here, ``implements AccessKeyId``

File: src/main/java/com/amihaiemil/charles/aws/StRegion.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ */
+package com.amihaiemil.charles.aws;
+
+/**
+ * Implementation for {@link SecretKey} interface.
+ * @author Sherif Waly (sherifwaly95@gmail.com)
+ * @version $Id$
+ * @since 1.0.0
+ */
+public final class StRegion extends AbstractSystemProperty implements SecretKey {
```
Comment (amihaiemil) [inline src/main/java/com/amihaiemil/charles/aws/StRegion.java]: @SherifWaly Here, ``implements Region``

**Comment by amihaiemil** [comment]
@SherifWaly As you probably noticed, interfaces ``Region``, ``SecretKey``, ``EsEndPoint``,  and ``AccessKeyId`` are not really needed, we could just have ``RtSystemProperty implements SystemProperty``. But I decided to let them be since they offer better semantics to the code.

For instance, when instantiating a [SignedRequest](https://github.com/opencharles/charles-rest/blob/master/src/main/java/com/amihaiemil/charles/aws/requests/SignedRequest.java) or an [EsHttpRequest](https://github.com/opencharles/charles-rest/blob/master/src/main/java/com/amihaiemil/charles/aws/requests/EsHttpRequest.java), you will clearly see that they expect a ``Region`` or an ``EsEndPoint`` precisely, rather than a generic ``SystemProperty``. Or, when you'll look at the ctor of [AwsCredentialsFromSystem](https://github.com/opencharles/charles-rest/blob/master/src/main/java/com/amihaiemil/charles/aws/requests/SignedRequest.java#L78) you will clearly see that it expects a ``SecretKey`` and an ``AccessKeyId``, rather than two ``SystemProperty``, which you will wonder what represent :D 

Also, we will have a few decorators, both for generic ``SystemProperty``, as well as particulars, like ``EsEndPoint``. 

Makes sense? What do you think?

**Review by amihaiemil — COMMENTED** [review]
@SherifWaly 2 more comments

File: src/main/java/com/amihaiemil/charles/aws/StEsEndPoint.java [hunk 5]
```diff
@@ -40,5 +40,13 @@
     public StEsEndPoint(final String name) {
         super(name);
     }
-
+ 
+    @Override
+    public String read() {
+        String endPoint = super.read();
+        if(!endPoint.endsWith("\\")) {
```
Comment (amihaiemil) [inline src/main/java/com/amihaiemil/charles/aws/StEsEndPoint.java]: @SherifWaly We need "/" (slash), not "\\" (backslash) :D Sorry, I might have written it wrong when I let the comment the first time.

**Review by amihaiemil — COMMENTED** [review]
@SherifWaly Looks good! just one more minor thing :D

File: src/main/java/com/amihaiemil/charles/aws/StEsEndPoint.java (hunk 5 above)
Comment (SherifWaly) [reply src/main/java/com/amihaiemil/charles/aws/StEsEndPoint.java]: @amihaiemil Ok, I've made a new commit :D

File: src/main/java/com/amihaiemil/charles/aws/StRegion.java (hunk 4 above)
Comment (SherifWaly) [reply src/main/java/com/amihaiemil/charles/aws/StRegion.java]: @amihaiemil Sorry for this mistake.

**Comment by SherifWaly** [comment]
@amihaiemil Please check last commit :)

**Comment by amihaiemil** [comment]
@rultor pls merge

**Pull request merged**